grpc-gateway/client/subscribeToEventsWithCurrentState.go

```diff
@@ -0,0 +1,197 @@
+	return c.subClient.Context()
+}
+
+// SendMsg is generally called by generated code. On error, SendMsg aborts
+// the stream. If the error was generated by the client, the status is
+// returned directly; otherwise, io.EOF is returned and the status of
+// the stream may be discovered using RecvMsg.
+//
+// SendMsg blocks until:
+//   - There is sufficient flow control to schedule m with the transport, or
+//   - The stream is done, or
+//   - The stream breaks.
+//
+// SendMsg does not wait until the message is received by the server. An
+// untimely stream closure may result in lost messages. To ensure delivery,
+// users should ensure the RPC completed successfully using RecvMsg.
+//
+// It is safe to have a goroutine calling SendMsg and another goroutine
+// calling RecvMsg on the same stream at the same time, but it is not safe
+// to call SendMsg on the same stream in different goroutines. It is also
+// not safe to call CloseSend concurrently with SendMsg.
+func (c *GrpcGateway_SubscribeToEventsClient) SendMsg(m interface{}) error {
+	req, ok := m.(*pb.SubscribeToEvents)
+	if !ok {
+		return fmt.Errorf("invalid type('%T')", m)
+	}
+	if req.GetCreateSubscription() != nil {
+		_, ok := c.subs.Load(req.GetCorrelationId())
+		if ok {
+			return fmt.Errorf("duplicit correlationID(%v)", req.GetCorrelationId())
+		}
+		_, ok = c.newSubs.Load(req.GetCorrelationId())
+		if ok {
+			return fmt.Errorf("duplicit correlationID(%v)", req.GetCorrelationId())
+		}
+		sub := NewSub(c.subClient.Context(), c.grpcClient, c.send, req.GetCorrelationId(), c.deduplicateExpiration, req.GetCreateSubscription())
+		c.newSubs.Store(req.GetCorrelationId(), sub)
+	}
+	if req.GetCancelSubscription() != nil {
+		c.removeSubs.Store(req.GetCancelSubscription().GetSubscriptionId(), true)
+	}
+	return c.subClient.SendMsg(m)
+}
+
+// RecvMsg blocks until it receives a message into m or the stream is
+// done. It returns io.EOF when the stream completes successfully. On
+// any other error, the stream is aborted and the error contains the RPC
+// status.
+//
+// It is safe to have a goroutine calling SendMsg and another goroutine
+// calling RecvMsg on the same stream at the same time, but it is not
+// safe to call RecvMsg on the same stream in different goroutines.
+func (c *GrpcGateway_SubscribeToEventsClient) RecvMsg(m interface{}) error {
+	for {
+		if len(c.events) > 0 {
+			ev := c.events[0]
+			c.events = c.events[1:]
+			event := m.(*pb.Event)
+			event.CorrelationId = ev.GetCorrelationId()
+			event.SubscriptionId = ev.GetSubscriptionId()
+			event.Type = ev.GetType()
+			return nil
+		}
+		start := time.Now()
+		err := c.subClient.RecvMsg(m)
+		if err != nil {
+			return err
+		}
+		timeDiff := time.Since(start)
+		event := m.(*pb.Event)
+		switch ev := event.GetType().(type) {
+		case *pb.Event_OperationProcessed_:
+			s, ok := c.newSubs.LoadAndDelete(event.GetCorrelationId())
+			if ok {
+				// we don't want to store events
```

Comment:
Event are stored though? I guess should read 'we don't want to store events with error'?

Reply:
removed